Makefile

```diff
@@ -3,10 +3,10 @@
 # to be set, but for the time being we need to override the defaults so that our existing dev environment works. This
 # can be removed when that is resolved.
 ifeq ($(CC),cc)
-CC = gcc-13
+CC = ccache /usr/bin/clang-18
```

Comment:
testing this locally and commenting out these out in my .bashrc:

```
export CC="ccache /usr/bin/clang-18"
export CXX="ccache /usr/bin/clang++-18"
```

still led to `make` using `g++13`
```
vagrant@expensidev2004:/vagrant/Bedrock$ make -j8
g++-13 -g -std=c++20 -fPIC -DSQLITE_ENABLE_NORMALIZE -O0 -Wall -Werror -Wformat-security -Wno-unqualified-std-cast-call -Wno-error=deprecated-declarations -I/vagrant/Bedrock -I/vagrant/Bedrock/mbedtls/include -MMD -MF .build/test/clustertest/testplugin/ExternPointer.d -MT .build/test/clustertest/testplugin/ExternPointer.o -o .build/test/clustertest/testplugin/ExternPointer.o -c test/clustertest/testplugin/ExternPointer.cpp
g++-13 -g -std=c++20 -fPIC -DSQLITE_ENABLE_NORMALIZE -O0 -Wall -Werror -Wformat-security -Wno-unqualified-std-cast-call -Wno-error=deprecated-declarations -I/vagrant/Bedrock -I/vagrant/Bedrock/mbedtls/include -MMD -MF .build/test/clustertest/testplugin/TestPlugin.d -MT .build/test/clustertest/testplugin/TestPlugin.o -o .build/test/clustertest/testplugin/TestPlugin.o -c test/clustertest/testplugin/TestPlugin.cpp
```

I believe b/c
```
vagrant@expensidev2004:/vagrant/Bedrock$ echo $CXX
g++-13
```

but I'm not exactly sure where that is getting set, maybe something in expensidev?

At least 👍 this build properly with clang on GH Actions 
<img width="1390" alt="image" src="https://github.com/user-attachments/assets/acb89196-574f-44fd-aee2-400eaaf9c5f9">

Reply:
Did you close the shell and reopen a new one after commenting those lines out? They will still be set in your existing shell and any new shells opened from it until you exit.